README.md

```diff
@@ -265,6 +265,14 @@ It's useful in several situations, e.g.:
 - Joining separate changes that make more sense when applied together
 - Rewriting _work in progress_ commits
 
+### When to avoid rebase or squash?
+
+Both rebase and squash commands substitutes existing commits for new ones. So it's not recommended to use with branches/commits that are being referenced by others or exists on the remote repository.
+
+Because when we are rebasing or squashing commits that anyone depends on and were not yet pushed to the remote repository, it will have no side effects. But, if the commits exist already on a remote repository and we try push our changes, git will identify that our commits diverge from the commits that exists on the repository and will fail requesting us to get those existing commits from the repository, but that would end of duplicating or maintaining the commits we tried change.
```

Comment:
What do you thikn think about these changes?

```diff
- Because when we are rebasing or squashing commits that anyone depends on and were not yet pushed to the remote repository, it will have no side effects. 
+ Rebase or squash of local commits (i.e., commits not pushed to remote repository) will not generate conflicts in a remote repo, since there is neither rewrite of remote history nor replacement of existing commits.
```

I think we can remove the `that anyone depends on` (since it is local and never were pushed, no one depends upon they) and replace `side effects` with the specific side effects.